src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/reader/CatalogItemReader.java

```diff
@@ -0,0 +1,86 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2016 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.backuprestore.reader;
+
+import java.util.Arrays;
+
+import org.geoserver.backuprestore.Backup;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.util.CloseableIterator;
+import org.geoserver.config.util.XStreamPersisterFactory;
+import org.opengis.filter.Filter;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.item.ItemReader;
+import org.springframework.core.io.Resource;
+
+/**
+ * Concrete Spring Batch {@link ItemReader}.
+ * 
+ * Reads resource items from in memory {@link Catalog}.
+ * 
+ * @author Alessio Fabiani, GeoSolutions
+ *
+ */
+public class CatalogItemReader<T> extends CatalogReader<T> {
+
+    CloseableIterator<T> catalogIterator;
```

Comment:
The reader is closable because it needs to be closed at the end, in JDBCConfig it will hold a JDBC connection for example